File: C++/762.cpp
__________________________________________________________________________________________________
sample 4 ms submission
class Solution {
public:
    int countPrimeSetBits(int L, int R) {
        const int mask = 665772;
        int ans = 0;
        for(int i=L; i<=R; i++){
            if(mask & (1 << __builtin_popcountll(i))){
                ans++;
            }
        }
        return ans;
    }
};
__________________________________________________________________________________________________
sample 8148 kb submission
static int speedup = [](){
    std::ios::sync_with_stdio(0);
    std::cin.tie(0);
    return 0;
}();

class Solution {
public:
    bool is_prime(int a){
        if(a == 2 || a == 3 || a == 5 || a == 7 || a == 11 || a == 13 || a == 17 || a == 19 || a == 23 || a == 29 || a == 31){
            return true;
        }
        return false;
    }
    
    int countPrimeSetBits(int L, int R) {
        int ans = 0;
        for(int i = L; i <= R; i++){
            int c = __builtin_popcount(i);
            if(is_prime(c)){
                ans++;
            }
        }
        return ans;
    }
};
__________________________________________________________________________________________________

File: Java/762.java
__________________________________________________________________________________________________
sample 1 ms submission
class Solution {
    public int countPrimeSetBits(int L, int R) {
        int count = 0;
        P[2] = true;
        P[3] = true;
        P[5] = true;
        P[7] = true;
        P[11] = true;
        P[13] = true;
        P[17] = true;
        P[19] = true;
        for (int i = L; i <= R; i++) {
            if(isPrimeSet(i)) count++;
        }
        return count;
    }
    
    boolean[] P = new boolean[20];
    
    boolean isPrimeSet(int x) {
        int c = Integer.bitCount(x);
        return P[c];    
    }
}
__________________________________________________________________________________________________
sample 31844 kb submission
class Solution {
    public int countPrimeSetBits(int L, int R) {
        int count = 0;
        for (int i=L; i<=R; i++) {
            if (isPrime(numberOfOnes(i))) {
                count++;
            }
        }
        return count;
    }
    
    public boolean isPrime(int num) {
        if (num <= 1) return false;
        int bound = (int) Math.sqrt(num);
       // boolean prime = true;
        for (int i=2; i<=bound; i++) {
            if (num % i == 0) {
                return false;
            }
        }
        
        return true;
    }
    
    public int numberOfOnes (int num) {
        
        int count = 0;
        while (num != 0) {
            num = num & (num-1);
            count++;
        }
        
        return count;
    }
}
__________________________________________________________________________________________________
